tests/cpp-empty-test/Classes/reader/CreatorReader.cpp

```diff
@@ -131,6 +131,10 @@ CreatorReader::CreatorReader()
 {
     _animationManager = new AnimationManager();
     _collisionManager = new ColliderManager();
+    _animationManager->autorelease();
+    _collisionManager->autorelease();
+    CC_SAFE_RETAIN(_animationManager);
+    CC_SAFE_RETAIN(_collisionManager);
```

Comment:
Why release and retain it? It does nothing.

Reply:
It do add the `_animationManager, _collisionManager ` to release pool,  so as to make release both objects in destruction function effective. Or both object will never released because they didn't existed in the release pool.

```
CreatorReader::~CreatorReader()
{
    CC_SAFE_RELEASE_NULL(_collisionManager);
    CC_SAFE_RELEASE_NULL(_animationManager);
}
```